woocommerce/woocommerce-admin
# Use version number for cache invalidation

Fixes #2020.

This PR seeks to do the following:
- Use transients to cache report data store queries
- Invalidate caches using a version number (incremented when data changes)

### Question:

Are there other places we need to invalidate cache?

### Detailed test instructions:

- Disable any persistent object cache plugins (makes testing easier)
- Open any Report
- Verify a transient containing `wc_report_[report name]` exists with a version number at the end (`:1`)
- Modify a user, order, or coupon
- Verify the `woocommerce_reports_cache_version` option value changes
- Reload the report
- Verify a new transient is created with the new version number appended



### Changelog Note:



Performance: add caching layer to analytics.
